packages/contracts-ops/contracts/scripts/DeployImplementations.s.sol

```diff
@@ -0,0 +1,254 @@
+            updateImpl(localDomain);
+            vm.stopBroadcast();
+        }
+    }
+
+    function loadConfig() internal {
+        recoveryTimelock = getRecoveryTimelock(localDomain);
+        xAppConnectionManager = address(getXAppConnectionManager(localDomain));
+        updaterManager = address(getUpdaterManager(localDomain));
+        localDomainNumber = uint32(getDomainNumber(localDomain));
+        title("Input");
+        console2.log("Timelock:          ", recoveryTimelock);
+        console2.log("XCnMngr:           ", xAppConnectionManager);
+        console2.log("UpdMngr:           ", updaterManager);
+        console2.log("DomNubr:           ", uint256(localDomainNumber));
+
+        // Input validation
+        require(localDomainNumber != 0, "localDomainNumber can't be zero");
+        require(recoveryTimelock != 0, "recovery timelock can't be zero");
+        require(
+            xAppConnectionManager != address(0),
+            "xAppConnectionManager can't be address(0)"
+        );
+        require(
+            updaterManager != address(0),
+            "updaterManager can't be address(0)"
+        );
+    }
+
+    function createForks() internal {
+        title("Loading the forks from the following RPCs..");
+        for (uint256 i; i < domains.length; i++) {
+            string memory rpcEndpoint = getRpcs(domains[i])[0];
+            console2.log(domains[i], "--->", rpcEndpoint);
+            forkIdentifiers.push(vm.createFork(rpcEndpoint));
+        }
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                      IMPLEMENTATION DEPLOYMENT
+  //////////////////////////////////////////////////////////////*/
+
+    function deployImplementations() internal {
+        // NOTE: these init values do not map to correct expected values.
+        // Storage variables in implementation contracts don't matter.
+        // Purpose is to initialize the implementations as a matter of best practice,
+        // despite the fact that in Nomad's architecture,
+        // un-initialized implementations can't harm the protocol
+        // (unless, in the future, we introduce delegatecall in any implementations)
+        // Home
+
+        title("Deployment Information");
+
+        home = new Home(localDomainNumber);
+        home.initialize(IUpdaterManager(updaterManager));
+        console2.log("Home deployed at", address(home));
+
+        // Replica
+        replica = new Replica(localDomainNumber);
+        replica.initialize(0, address(0), bytes32(0), 0);
+        console2.log("Replica deployed at", address(replica));
+
+        // GovernanceRouter
+        governanceRouter = new GovernanceRouter(
+            localDomainNumber,
+            recoveryTimelock
+        );
+        governanceRouter.initialize(xAppConnectionManager, address(0));
+        console2.log("Governance Router deploed at", address(governanceRouter));
+
+        // BridgeRouter
+        bridgeRouter = new BridgeRouter();
```

Comment:
what about `EthereumBridgeRouter` for ethereum?

Reply:
@odyslam if we wanted to merge this at some point, would have to take this into account. but by now, i think we should move forward with my modifications of your work, as it's closer to mergeable. you can refactor / edit that work (And the other scripts) to make them multi-chain and use `writeJson` - but we shouldn't block timelines on that.